### PR TITLE
Get data from API and convert the Search Page to be dynamic 

### PR DESCRIPTION
**Pull request by AFahgan**

### CODE REVIEW

**Review by AhmedQeshta — CHANGES_REQUESTED** [review]

File: client/src/Components/SearchPageComponents/CardResultFilter/index.jsx [hunk 1]
```diff
@@ -20,6 +21,15 @@ export default function CardResultFilter() {
     <div className="card_result_filter__container">
       {filter.length !== 0 &&
         filter.map((data) => <Card key={data.id} filter={data} />)}
+      <div className="switchdiv">
+        <Pagination
+          count={10}
```
Comment (AhmedQeshta) [inline client/src/Components/SearchPageComponents/CardResultFilter/index.jsx:26]:
```suggestion
          count={filter.length} // or **arr length**
```

File: client/src/Store/Services/gyms.js [hunk 2]
```diff
@@ -12,7 +12,14 @@ export const gymsApi = createApi({
     getGymData: builder.query({
       query: (id) => `/${id}`,
     }),
+    getFillterData: builder.query({
+      query: (name) => `/filter?name=${name}`,
```
Comment (AhmedQeshta) [inline client/src/Store/Services/gyms.js:16]:
```suggestion
      query: (name) => `/filter?name=${name}&....`, // you must complete filter query
```

**Review by AhmedQeshta — APPROVED** [review]

**Review by rawaneyad — APPROVED** [review]
Great Job Ayman 👍🏼

File: client/src/Components/SearchPageComponents/Filter/index.jsx [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
+import React, { useEffect } from "react";
+import axios from "axios";
+import FilterSide from "../FilterSide";
 import CardResultFilter from "../CardResultFilter";
-
 import "./style.css";
 
 export default function Filter() {
+  const [filter, setFilter] = React.useState([]);
+
```
Comment (LinaYahya) [inline client/src/Components/SearchPageComponents/Filter/index.jsx:9]: add a state for pages
```suggestion
  const [page, setPage] = useState(0)
```
and then set count as `Math.ceil(page/3)`

File: client/src/Components/SearchPageComponents/Filter/index.jsx [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+  const changeFilterQuery = async (query = "") => {
+    const response = await axios(`/api/v1/gyms/filter?${query}`);
+    setFilter(response.data.gyms);
+  };
+
+  useEffect(() => {
+    changeFilterQuery();
+  }, []);
```
Comment (LinaYahya) [inline client/src/Components/SearchPageComponents/Filter/index.jsx:17]:
```suggestion

```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**